Mlem/App/Views/Shared/Navigation/LoginPage.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  LoginPage.swift
+//  Mlem
+//
+//  Created by Sjmarf on 13/05/2024.
+//
+
+import MlemMiddleware
+import SwiftUI
+
+enum LoginPage: Hashable {
+    case pickInstance
+    case instance(_ instance: any Instance)
+    case reauth(_ userStub: UserStub)
+    case totp(url: URL, username: String, password: String)
+    
+    @ViewBuilder
+    func view() -> some View {
+        switch self {
+        case .pickInstance:
+            LoginInstancePickerView()
+        case let .instance(instance):
+            LoginCredentialsView(instance: instance)
+        case let .reauth(userStub):
+            LoginCredentialsView(userStub: userStub)
+        case let .totp(url, username, password):
+            LoginTotpView(url: url, username: username, password: password)
+        }
+    }
+    
+    static func == (lhs: LoginPage, rhs: LoginPage) -> Bool {
+        switch (lhs, rhs) {
+        case (.pickInstance, .pickInstance):
+            true
+        case let (.totp(url1, username1, password1), .totp(url2, username2, password2)):
+            url1 == url2 && username1 == username2 && password1 == password2
+        case let (.instance(instance1), .instance(instance2)):
+            instance1.actorId == instance2.actorId
+        case let (.reauth(user1), .reauth(user2)):
+            user1 == user2
+        default:
+            false
+        }
+    }
+    
+    func hash(into hasher: inout Hasher) {
+        switch self {
+        case .pickInstance:
+            hasher.combine(0)
+        case .totp:
+            hasher.combine(1)
+        case let .instance(instance):
+            hasher.combine(instance.actorId)
+        case let .reauth(user):
+            hasher.combine(user.actorId)
+        }
+    }
```

Comment:
We're inconsistent on this in 1.0, but I think long-term it'll serve us better to hash string descriptions rather than an incrementing int--there's *possibly* a very tiny performance hit, but this lets us do things like reorder cases or add new cases without either putting the numbers out of order (oh, the humanity) or updating them; it also lowers the chance of accidentally creating a duplicate hash value.

```suggestion
    func hash(into hasher: inout Hasher) {
        switch self {
        case .pickInstance:
            hasher.combine("pickInstance")
        case .totp:
            hasher.combine("totp")
        case let .instance(instance):
            hasher.combine(instance.actorId)
        case let .reauth(user):
            hasher.combine(user.actorId)
        }
    }
```